wasmtime-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -305,7 +301,7 @@ bool fd_prestats_insert(
 }
 
 // Looks up a preopened resource table entry by number.
-static __wasi_errno_t fd_prestats_get_entry(
+__wasi_errno_t fd_prestats_get_entry(
```

Comment:
I reckon this function could easily be re-implemented in Rust, and should be done in this PR. Would you agree?

Reply:
Yeah, this is a relatively simple one.

Reply:
OK, I've implemented it as a safe fn in Rust. Have a look and let me know what you think.